### DISCUSSION TITLE
guest party mode / jukebox

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#411] [Feature requests and ideas]
Let's say you have a party and you want your guests to have control over the music. Just like an old-school jukebox.
Globally my idea is as follows:

- Create some option/button to enable "guest/party mode" on a Queue.
- Have some easy way for guests to control the queue, for example a separate web URL and/or simple app.
- It must be plain simple, for example have a QR code or other share code to share the guest login with your guests
- Once the party is over you withdraw the party mode.
- Guests can add songs to the queue
- You (as host) select the main playlist and from which music the guests may pick (default to all music from all providers)
- You (as host) select the add mode in which way tracks are added to the queue. For example FIFO which will place each guest request after the current playing track (but in order of arrival) or random which will insert the tracks somewhere within the queue or "dj mode" where the track is intelligently inserted in the queue based on surrounding music and number of requests
- Guests are not able to do volume control on the queue ?
- Guests are not able to pause/play the music ?

**Comment by nagyrobi** [comment]
+1 for FIFO!
- Guests are not able to skip/next/previous/seek tracks
- don't require installing any other new app for this, should be as simple as possible URL displayed in mobile browser
- the QR code picture of the URL should be accessible as an attribute or an image entity so we can display it on standard Lovelace or any other UI (eg. Picture Entity card or via [OpenHASP](https://www.openhasp.com) display)
- when the party is over have an option to save the Queue for later reuse
- optionally serve the Jukebox UI from a separate port number of HA in order to be able to filter access via guest-network firewalls (eg. prevent access to main HA UI, but still allow access to Jukebox on a separate tcp port)
- some kind of binary sensors to show that `party mode` is enabled and separately that playback is done from the party mode queue itself. useful to trigger other stuff in Home Assistant like change lights, set covers position, change Lovelace theme etc.
- also reverse, a service call to start party mode on an external trigger (eg. when special person arriving home, start party mode)

**Comment by erkr** [comment]
Nice idea! To add:
- allow the host to author the queue (move tracks up/down, or remove)

**Comment by majbom** [comment]
**Very nice idea!**

Maybe add:

- some sort of spam protection to prevent guests adding tons of tracks

**Comment by patienttruth** [comment]
This would be awesome. +1 vote

All that said, in the interim, being able to hide the settings tab from the front end when exposed to an external port would be a good stepping stone.

Since I don't have a lot of exposure on my home network I will expose this locally for guests to use, however I'd prefer if someone couldn't accidentally change a setting. ~~Especially since I haven't figured out how to backup the add-on's config.~~ I was able to create a backup via HA backup where I can create a "partial" backup of just this add-on which is good for now. Pretty low risk...

**Reply by OzGav** [reply]
Here is one approach you could try https://github.com/orgs/music-assistant/discussions/2175

**Reply by patienttruth** [reply]
That's awesome looking. Thanks!

**Comment by w00master** [comment]
My #1 feature request!  In terms of "self-use" of MA, I find that I have "most of what I need" - sure there are some small things here and there, but actually having a "Jukebox" or "Party" mode would be the biggest thing I'm looking for now with MA.

**Comment by fsmythe** [comment]
This would be great! Just a simple web interface with the ability to add to a currently running queue would be perfect.

**Comment by DJS91** [comment]
Hey All, 

I've been working on a Jukebox solution for MA.
I only have Apple music provider so it will need testing with other providers although it should work fine.
Let me know how you go with it

https://github.com/DJS91/HAMusicAssistantJukebox

@marcelveldt The latest build should now satisfy just about all of your listed requirements :D

**Reply by marcelveldt** [reply]
Super cool, Nice work!

**Reply by marcelveldt** [reply]
Small question: why choose to make it an HA integration and use HA in between ?
This could potentially also be a MA plugin provider (even with its own web interface).
PM me if you'd be interested in migrating it to a MA plugin provider, then I'll help you to get that moving.

**Reply by DJS91** [reply]
To be honest, it's my first dive into making anything for home assistant and initially, it was just a collection of files with a detailed instruction of where to place/customise them to make it work.
Then someone kindly offered to help make it a custom component for an easier install, so it became one.
It's been a good learning experience.
A plugin provider would be nice! I'll shoot you an email.

**Comment by maxi1134** [comment]
I see that the new party mode has been implemented in [2.8.0 BETA 18](https://github.com/music-assistant/server/releases/tag/2.8.0b18)

I have been playing for it for a few minutes and was wondering how I could load the page with the QR code on a browser without human interaction.
After trying to load 'http://192.168.0.15:8095/#/party', seeing that this was the URL shown on my FKB when the QR code is present, i realized that this gets redirected to 'http://192.168.0.15:8095/#/guest' automatically when loaded with the `fully_kiosk.load_url` action call.

It would also be great if we can load this URL from a tablet who is logged into MA as a user and not an admin!

**Reply by apophisnow** [reply]
If you are getting redirected to the /guest page, then I believe you at some point scanned or clicked the qr code which has logged you in with the guest account (which gets redirected to the guest page only). You will most likely need to clear your browser data (token stored in local storage) and log back in again.

In order to open the /party page (dashboard) without human interaction, I recommend creating a user account, and then a long lived token for that user. You can then log in by providing the token in the query string.

**Reply by DJS91** [reply]
What would the dashboard URL with the token as a query string look like?

**Reply by apophisnow** [reply]
You can right click the qr code and copy the link which includes the join code. It should look something like http://hostname:8095/?join=3BE8CC if you do not have remote access enabled. If you do have remote access enabled, then it will include a remote access token as well, and be pointing to music-assistant.io.

So you would need to add /beta and end up with a URL similar to "https://app.music-assistant.io/beta/?remote_id=SOMEID&join=SOMECODE"

**Reply by DJS91** [reply]
> In order to open the /party page (dashboard) without human interaction, I recommend creating a user account, and then a long lived token for that user. You can then log in by providing the token in the query string.

Apologies, I was referring to where to put the long-lived access token as a query string param to reach the /#/party dashboard?

I assume something similar to? 
http://homeassistant:8123/api/hassio_ingress/INGRESSID/#/party?token=LONGLIVEDACCESSTOKEN

**Reply by OzGav** [reply]
As per here https://www.music-assistant.io/faq/how-to/#access-the-ma-views-directly-via-url

**Reply by DJS91** [reply]
trying 
http://homeassistant:8095/?code=TOKEN#/party

it seems to redirect to #/guest instead of #/party

**Reply by maxi1134** [reply]
I have tried many things ( Code=Token# is among them), but I am unable to easily show the '#party' endpoint on a tablet.

Ideally, we would have a static URL we can simply load on a tablet, and it would show the party mode.

Having to manually open it on 10 tablets kinda defeats the point of automation in my opinion.

**Reply by apophisnow** [reply]
I'm curious where the failure is happening. The party mode display needs access to the queue and things from Music Assistant, so it has to authenticate, not sure exactly what you mean by a static URL in this case, but without auth it won't work. If you create a long lived token (NOT based on the built in party mode guest account that party mode itself creates), you can use it to put a display on the /party page. I have tested this with a remote wall display. The token must be created on a user account that has permission to the player that you have configured for party mode.

**Reply by maxi1134** [reply]
The failure is happening at the automation level.

I simply cannot store a token in my 'Automations.yaml' as I publicly share this one to help new users who are looking for examples of a complex config.

I have been trying, for an hour now, to get the token stored in two different input_text sensors, as I need to keep the token in the `!secret` file, and the length limit of the `input_text` sensor is set at 255 characters (The token length is ~370).

Unfortunately, I just learned that you cannot use variables in the `fully_kiosk.load_url action like so:

```
action: fully_kiosk.load_url
metadata: {}
data:
  device_id: 3855559a40e217d20c661fd5e6e1b452
  url: http://192.168.0.15:8095/?code='{{states(input_text.ma_jukebox_token_1}}''{{states(input_text.ma_jukebox_token_2}}'#/party

```
Ideally, accessing `http://192.168.0.15:8095/#/party` locally would open the page, without the need for an authentification.
I feel like having an authentication in front of a locally hosted page that is meant to be public is counterproductive.

Anyone who plans on hosting events and using this Jukebox will need to have a way to easily display the QR code at the very least.
Not only for the initial opening of the page, but for its reloading if someone exists it on the kiosk. 
(`If url ~= */party then reload`)

I will continue toying with the automations to try and get the QR/Queue page to appear automatically on tablets and add a comment with any possbible solution I might find!

**Reply by DJS91** [reply]
> The failure is happening at the automation level.
> 
> 
> 
> I simply cannot store a token in my 'Automations.yaml' as I publicly share this one to help new users who are looking for examples of a complex config.
> 
> 
> 
> I have been trying, for an hour now, to get the token stored in two different input_text sensors, as I need to keep the token in the `!secret` file, and the length limit of the `input_text` sensor is set at 255 characters (The token length is ~370).
> 
> 
> 
> Unfortunately, I just learned that you cannot use variables in the `fully_kiosk.load_url action like so:
> 
> 
> 
> ```
> 
> action: fully_kiosk.load_url
> 
> metadata: {}
> 
> data:
> 
>   device_id: 3855559a40e217d20c661fd5e6e1b452
> 
>   url: http://192.168.0.15:8095/?code='{{states(input_text.ma_jukebox_token_1}}''{{states(input_text.ma_jukebox_token_2}}'#/party
> 
> 
> 
> ```
> 
> Ideally, accessing `http://192.168.0.15:8095/#/party` locally would open the page, without the need for an authentification.
> 
> I feel like having an authentication in front of a locally hosted page that is meant to be public is counterproductive.
> 
> 
> 
> Anyone who plans on hosting events and using this Jukebox will need to have a way to easily display the QR code at the very least.
> 
> Not only for the initial opening of the page, but for its reloading if someone exists it on the kiosk. 
> 
> (`If url ~= */party then reload`)
> 
> 
> 
> I will continue toying with the automations to try and get the QR/Queue page to appear automatically on tablets and add a comment with any possbible solution I might find!

This may help

I have two smart tv's and an android tablet around my house that i pre-logged into (keep logged in checkbox) a dedicated HA user account i created called MAParty. I visited and logged into music assistant on each device once, which logs into music assistant and creates a session in the browser. Now i can open my static http://192.168.1.34:8123/api/youringress/#/party url in the browser and the party mode will load using the session auth.

I have an automation that launches the browser app on each device with the static /#/party url as the preconfigured homepage to launch the dashboard.

This was all set up yesterday.
Works great so far.
Will see how the token refresh goes after 8 hours of inactivity though?

Unsure if HA keep me logged in checkbox extends to MA auth or not. 
If it forces me to manually re-login to MA after some time of inactivity, this would remain an issue.

**Reply by apophisnow** [reply]
Thanks so much for the awesome detail! I'll try to find a good solution for this as I dig more into the HA integration side of things.

**Comment by hundredfire** [comment]
Hello! Happy to see this feature implemented, this is something that I've been searching for, for a while. I just updated to 2.8.0b18 to test it but I'm not sure how to actually access the feature? I can't see any mention of party mode in the interface.
I'm guessing the documentation is being worked on rn, but could someone give me some pointers? 
For info, I installed MA on HAOS (on a raspberry pi). I have home assistant accessible through the internet via a cloudflared tunnel, although I guess this doesn't matter

**Reply by maxi1134** [reply]
> Hello! Happy to see this feature implemented, this is something that I've been searching for, for a while. I just updated to 2.8.0b18 to test it but I'm not sure how to actually access the feature? I can't see any mention of party mode in the interface. I'm guessing the documentation is being worked on rn, but could someone give me some pointers? For info, I installed MA on HAOS (on a raspberry pi). I have home assistant accessible through the internet via a cloudflared tunnel, although I guess this doesn't matter


It's under providers in settings!

**Reply by hundredfire** [reply]
Duh! I should have looked harder. Thanks to the team and @apophisnow ! It works a treat and looks gorgeous!
I noticed that the feature collides with the beta remote access feature. If remote access is enabled, scanning the QR code redirects me to the remote access login portal and then to the home assistant main interface. Disabling remote access solves this. 
I personally don't need remote access so it's not a biggie. Just a heads-up

**Comment by DJS91** [comment]
@apophisnow Party mode plugin looks awesome! I will retire my repo and point folks to this.

Would it be possible to have a toggleable option for users who aren't on my Wi-Fi by having it accessible via a QR code with the external HA URL? 
When enabled, users accessing the queue via the external URL will need to enter an access code or private password to reach it.

**Reply by OzGav** [reply]
https://beta.music-assistant.io/plugins/party-mode/#remote-access

**Reply by DJS91** [reply]
ah, amazing. I see, clearly I haven't looked too closely at it yet haha

I enabled remote access, scanned the Party QR code, entered the remote access code when prompted, and i assumed it would take me me to the queue but seems to always prompt for ha login.

Using private browser session
I'm guessing this is the same issue @hundredfire is having?

**Reply by apophisnow** [reply]
By the way Marvin fixed the issue with remote access, so this should be working as expected now!

**Comment by apophisnow** [comment]
It isn't pointing to /beta but if you add that to the url manually it
should work. And it should work when actually deployed and no longer needs
/beta

**Reply by DJS91** [reply]
AH, of course! Awesome effort mate! Thanks for doing what I should have done from the start haha

**Reply by apophisnow** [reply]
Thanks! I can't believe i wasn't aware of this thread at all until now.

**Comment by apophisnow** [comment]
Ah, yes that should still work as described, so use "code=xxx" where xxx is
your token.

**Comment by apophisnow** [comment]
Then you probably already have a guest token in your browser local storage.
You would need to clear your browser data.

**Reply by DJS91** [reply]
I'm using private browsing sessions to test so there's no cache, I assume this should be fine?

**Comment by apophisnow** [comment]
Did you generate the token from the party guest user account then? You
should use a regular user account if you want to access anything other than
the guest page. The party mode guest user is a system created account that
is locked to the guest page.

**Reply by DJS91** [reply]
Ah ok i understand now. will use a HA account to auth to the party dash. An MA user token is just for authing to the guest queue view

**Comment by apophisnow** [comment]
I'm not super familiar with how the HA interaction works, but outside of HA
you could create a different MA user account for this purpose of you wanted
to. It may be possible to do the same with the HA integration, but I
wouldn't know.

**Reply by DJS91** [reply]
no worries. fyi, your replies are created as new comments instead of replies on the original comment. if you want to delete them all. no worries.

**Comment by lanthaler** [comment]
It would be great if there was a configuration option that would allow to show the QR code on the cast receiver app when in party mode. I typically cast music to my Android TV or smart display and it would be great to allow guests to directly join the party from there :-)

@apophisnow are you already working on that or are there any other plans to get this onto a screen easily?

**Reply by apophisnow** [reply]
Can you explain your setup a bit more? I'm not familiar with how you would be casting exactly?

**Reply by lanthaler** [reply]
Music Assistant already has a Chromecast provider that adds tht respective players. Playing music on an Android TV or smart display shows as Music Assistant receiver app which shows title, cover art etc. ([screenshot](https://www.music-assistant.io/blog/2025/12/17/music-assistant-2-7/#rebel-yell))

Party mode requires to select the players to control. I was thinking that if that player was a Chromecast player the receiver app could be updated to show the QR code directly so that guests can easily join.

**Reply by apophisnow** [reply]
Ah ok. I will have to look into that. I haven't tried the chromecast provider. Great idea though, thanks!

**Reply by lanthaler** [reply]
Thanks a lot for your work on Party! It will be one of the most differentiating features for Music Assistant

**Comment by DJS91** [comment]
<img width="1195" height="935" alt="Screenshot 2026-03-16 140330" src="https://github.com/user-attachments/assets/c1674fa1-ef1e-49ee-b8fe-f71cfa6a7667" />

Can a progress bar/playtime be added to the current box?

**Reply by apophisnow** [reply]
It could be, but I'm not sure what value it would bring. The party dashboard isn't an area where you would be scrubbing through the track, so it wouldn't be functioning as a control. My preference would be to keep it clean and minimal. Hopefully at a party one is not staring at this page longer than it would take to answer the question of "Hey, I like this song, what is it?" or "What was that last song that was playing?" or "What song is coming up next?"

I'm curious what your thoughts are.

**Reply by DJS91** [reply]
Had a situation where a guest found and queued a really long track and nobody else could tell how long it would be until the next one.

The intention is not for scrubbing or interaction, just a subtle visual indicator of song progress so guests can get an idea of when the current song will finish and when to expect the next one.

I think it would also help during lyrics mode and karaoke so you can keep track of/remember where you are for singing during a track.

hopefully that makes sense?

**Reply by apophisnow** [reply]
Yeah, that makes sense. I suppose as an optional setting I wouldn't be opposed to it. Thanks for the suggestion!

**Comment by olterman** [comment]
## Add delay setting for karaoke mode 

I don't know if I can write this here or if I need to make a new feature request but ....
For those of us that use snapcast the karaoke mode is hopelessly out of sync since there is a set delay on all speakers, 
I would love if there was a delay setting in the karaoke section that allows for setting a manual delay for the lyrics. 

#cheers

**Comment by anthonws** [comment]
May I kindly ask for an option (or by default), to not open a new page when one clicks "Party". It doesn't work well with things like Fully Kiosk and alike. It should just show inline, like everything else.

TY!

**Reply by apophisnow** [reply]
Planning to implement this behavior by default, and have an advanced option for those that prefer opening a separate tab. Thanks for the input!

**Comment by AXEBro** [comment]
This was an awesome feature - beautifully implemented.

Latest betta 0b22 - now only shows Music Assistant under the QR code. 

I'd like to add a line below that to say something like Join the party or Jukebox, etc. 

I love MA and tell the world about it - but on this version, someone might think the QR code is to download Music Assistant as opposed to controlling the music. I can completely see someone saying "why would I need that for, I already have Tidal/Spotify/Apple music"?

**Reply by apophisnow** [reply]
This is how it originally worked, but the team wanted to add the logo, and that feature was removed. Let me talk with them and see if we can find a different solution for the logo. @MarvinSchenkel

**Reply by OzGav** [reply]
I would say the UI doesn’t exist in a vacuum and someone would say you scan this to get access to the guest interface.

**Reply by apophisnow** [reply]
> I would say the UI doesn’t exist in a vacuum and someone would say you scan this to get access to the guest interface.

Also very true.

**Reply by AXEBro** [reply]
Yeah - I agree - it doesn't - but when actually hosting a party - people come and go. I might be in the yard, someone might be at the bar where the music is playing. 

I love MA - and everyone and of course want everyone and their mother to know about it. I would still want the MA info there, just another 1 liner underneath to add a message, like on the previous build. 

Just a thought. Either way - this was a fantastic release.

**Reply by apophisnow** [reply]
Regardless of where we go with it, I certainly appreciate the feedback! Thanks for being a part of it!

**Reply by AXEBro** [reply]
> Regardless of where we go with it, I certainly appreciate the feedback! Thanks for being a part of it!

Thank you for all guys have done w. Music Assistant. I cannot use anything else now. You've commoditized music access -if that makes sense. 

This was an glorious release.

**Reply by maxi1134** [reply]
As a weekly party host, when you start to have friends of friends of friends showing up, they don't always feel comfortable interacting with every guest-facing function. 
Having a customizable friendly text to nudge them to scan this and help set the ambiance/playlist is quite useful in those instance;
'Don't be shy, scan me and jam-on!`

Could we instead integrate the MA logo inside the QR code as many do?
And restore the custom text function in lieu?


<img width="289" height="289" alt="image" src="https://github.com/user-attachments/assets/a99b7ac2-7e61-4b0c-9f04-f192d740f006" />



And the mobile page could then have a Clickable 'Provided by Music Assistant' link somewhere at the bottom/top

**Reply by AXEBro** [reply]
> As a weekly party host, when you start to have friends of friends of friends showing up, they don't always feel comfortable interacting with every guest-facing function. Having a customizable friendly text to nudge them to scan this and help set the ambiance/playlist is quite useful in those instance; 'Don't be shy, scan me and jam-on!`

This guy parties. Can I be your friend of a friend?

**Reply by OzGav** [reply]
> Could we instead integrate the MA logo inside the QR code as many do?

No as that violates the QR code standard apparently. (It was discussed)

**Reply by maxi1134** [reply]
> > Could we instead integrate the MA logo inside the QR code as many do?
> 
> No as that violates the QR code standard apparently. (It was discussed)

https://en.wikipedia.org/wiki/QR_code#Frame_QR

https://www.qrcode.com/en/codes/frameqr.html

Link to patent holder website: https://www.denso-wave.com/en/system/qr/fundamental/qrcode/frameqr/index.html

It seems like a specific kind of QR standard exists for this.

**Comment by maxi1134** [comment]
I got to try this out during an actual party this weekend with around six guests using the jukebox. Here's what feedback came up from them, plus one idea of my own.

**Queue Spamming**
A few guests noticed it's pretty easy to spam the queue with the same song on repeat. It might be worth adding some kind of limit, either per client or a global cap that prevents the same track from appearing back-to-back beyond a set threshold.

**Boost Behavior on Own Tracks**
Some guests were confused that boosting a track they just queued added a second copy instead of promoting the one already in the queue. It would feel more natural if boosting your own queued track just moved the existing entry up rather than duplicating it.

**Suggestion: Vote to Skip**
Right now, skip power requires only one vote, by one person, which can feel a bit heavy-handed in a group setting. A vote-to-skip system could work well here, by either requiring a percentage of connected clients or a fixed number of votes before a track gets skipped. Would help keep things fair without taking control away from everyone else.

Other than this, they love the concept of having a jukebox!

**Reply by AXEBro** [reply]
Awesome points... 

I think something cool would be a static QR code that could be assigned for each player/group. I'd love to just print out a QR code and leave it by the bar.

**Reply by MarvinSchenkel** [reply]
@apophisnow you should be able to boost tracks when you are browsing the queue right? Boosting a song via search adds it again to the queue.

**Reply by maxi1134** [reply]
> @apophisnow you should be able to boost tracks when you are browsing the queue right? Boosting a song via search adds it again to the queue.

This is what my friends meant.

They would queue it in the search list, and then see the 'boost', and press it. Threfore leaving the song twice in the queue.

Ideally, if user X presses Queue on Y and then user X Boosts on Song Y, it would be moved up the queue.
But if User Z boosts song Y from HIS search bar, the one queued by user X would be unaffected.

I am also unsure if this is already a thing, but people should be able to boost other people's songs.
Example would be: I see something I like in the queue that someone else put, I boost it with my boost without their boost being used.

**Reply by apophisnow** [reply]
A lot of good ideas. Unfortunately, the way the plugin currently works is by having all users share a single guest account. So, most of these ideas can't really be implemented until we change how that part of things work. Here are my specific thoughts on your suggestsions.

1. Queue Spamming - By using the Party plugin you are giving guests access to play music on your system. I see this as more of a people problem, not a tech problem. If you have annoying party guests, perhaps ask them to knock it off, or leave in the worst case. The current implementation is not designed to protect against malicious users, and I'm not convinced it really needs to be.

2. Boost behavior on own tracks - User confusion is something I can get more behind fixing. Perhaps a setting to disallow duplicate tracks could be a partial solution (and might help with item one as well). Alternatively, perhaps disabling the buttons on a track that was just added or boosted (force the user to search again if they want to queue it twice).

3. Vote to skip - I like this idea, but we run into the problem of the shared account. I think it would perhaps be nice if each user got their own guest account, and could add their name even (but not need to make a password). Then we could do things like this, and show who requested which song etc.

**Reply by apophisnow** [reply]
> > @apophisnow you should be able to boost tracks when you are browsing the queue right? Boosting a song via search adds it again to the queue.
> 
> This is what my friends meant.
> 
> They would queue it in the search list, and then see the 'boost', and press it. Threfore leaving the song twice in the queue.
> 
> Ideally, if user X presses Queue on Y and then user X Boosts on Song Y, it would be moved up the queue. But if User Z boosts song Y from HIS search bar, the one queued by user X would be unaffected.
> 
> I am also unsure if this is already a thing, but people should be able to boost other people's songs. Example would be: I see something I like in the queue that someone else put, I boost it with my boost without their boost being used.

There is a toast notification when the song is added. Were they not noticing it?

**Reply by maxi1134** [reply]
> > > @apophisnow you should be able to boost tracks when you are browsing the queue right? Boosting a song via search adds it again to the queue.
> > 
> > 
> > This is what my friends meant.
> > They would queue it in the search list, and then see the 'boost', and press it. Threfore leaving the song twice in the queue.
> > Ideally, if user X presses Queue on Y and then user X Boosts on Song Y, it would be moved up the queue. But if User Z boosts song Y from HIS search bar, the one queued by user X would be unaffected.
> > I am also unsure if this is already a thing, but people should be able to boost other people's songs. Example would be: I see something I like in the queue that someone else put, I boost it with my boost without their boost being used.
> 
> There is a toast notification when the song is added. Were they not noticing it?

Maybe at first when they were sober 😅 But this is a party mode. And like in most parties, people drink and become the worst kind of users, inattentive and hurried.
I personally had to rework many of my tablet interfaces a few times to accommodate this. 

Notifications are also something that the common user ignores, we are ITs/Devs, so we know they hold crucial information, but after working over 10 years with users, I can guarantee that 3/4 of those notifications go ignored. (Not just in MA, I mean ALL notifications)

Edit: This is from a very technical friend who was trying out the Jukebox: 
<img width="411" height="95" alt="image" src="https://github.com/user-attachments/assets/96b5c4f5-7f25-420c-96e0-d4106e22612c" />
If he, a developer/sysadmin did not saw/noticed the notification, I don't expect my artists guests to do so.

**Reply by apophisnow** [reply]
Totally, part of the reason I made the link copied notification when you click the QR code pop up right in the middle of the QR code. Get the notification where they eyeballs are.

Can they elaborate on the expectation (how TouchTunes works)? I'd like to know the specific behavior they are looking for. Should boost not exist on the search, so you need to add a song first, then boost it from the queue if you want to play it sooner?

**Reply by maxi1134** [reply]
> * Queue Spamming - By using the Party plugin you are giving guests access to play music on your system. I see this as more of a people problem, not a tech problem. If you have annoying party guests, perhaps ask them to knock it off, or leave in the worst case. The current implementation is not designed to protect against malicious users, and I'm not convinced it really needs to be.
> 
> * Boost behavior on own tracks - User confusion is something I can get more behind fixing. Perhaps a setting to disallow duplicate tracks could be a partial solution (and might help with item one as well). Alternatively, perhaps disabling the buttons on a track that was just added or boosted (force the user to search again if they want to queue it twice).
> 
> * Vote to skip - I like this idea, but we run into the problem of the shared account. I think it would perhaps be nice if each user got their own guest account, and could add their name even (but not need to make a password). Then we could do things like this, and show who requested which song etc.

Sorry, just saw this preceding message!

>Queue Spamming 

This can actually not be malicious; it happened a few times due to my slow network (I was still scanning my songs). 
They would click 'Queue' a few times, as it did not react immediately. But yeah, I get what you mean by 'training' users, my problem is that I only know half the people in my house most times 😅

> * Boost behavior on own tracks

Disabling duplicated tracks, or having a 'minimum space' between them would definitely work.

> Vote to skip 

We don't even need names tbh. We could use the host-name/IP to mark people as 'different users'
If someone is going out of his way to switch his IP... Let him boost an extra song
And those can be dropped after 24 hours, as not to pollute the instance with 60 accounts.

**Reply by Obscuretone** [reply]
Hello! I'm the comrade. Also a very avid user of TouchTunes which is a commercial jukebox similar to this.

On the first part friends _are_ an issue that can be solved with tech. We do this all the time in bars, play the same song over and over until someone complains. Last was danger zone while a stag party was there :D 

I shouldn't be able to queue the exact same song file that's already in the queue. If there's multiple copies of a song, that's beyond the scope imho.

I would argue there's 4 states to a song. 

- not in the list at all
- in the queue by the auto-player (cool feature, btw)
- in the queue played by a user
- in the queue boosted by a user

Each step probably should be a promotion, that is the same song can only exist in one of these states. If it's played, the boost takes it over. If it's in the auto queue, the user queuing takes it over. 

If you're interested, I had codex take a crack at it, it seemed like a really simple change.

**Reply by Obscuretone** [reply]
Another note, I noticed DJ mode but it was unclear-ish what was happening. I thought it was cool, but I also thought i was imagining things. 

It could mark DJ songs _as_ DJ songs just for my sanity

**Comment by DJS91** [comment]
Hey @apophisnow 
Unsure if this is a bug or feature in the latest build but the /party dashboard page now shows the player controls and the MA sidebar by default.

For those of us that launch this on TV's and displays dedicated to the party dash, it would be great to have a switch/option to launch in full screen (or hide these) by default instead of having to navigate to launch full screen every time the page is loaded or refreshed.

**Reply by apophisnow** [reply]
Yeah there were a bunch of last minute changes. It needs some work to iron some things out. In the meantime, you may have good luck launching it directly with the frameless=true query parameter.

**Comment by xitters** [comment]
@apophisnow 
My party mod is skipping songs; it doesn't give any error message, it just skips them. What could be the problem? I'm using YouTube Music, and everything is updated to the latest version, 2.8.2. I think it started in version 2.8.1; I don't remember it happening before then.
It's random. I just tried adding songs when one was about to finish, and it skipped the three I added below it. It's weird.

**Reply by OzGav** [reply]
Try adding the exact same tracks to the queue directly (I.e. not via the party interface)  and see what happens then.

**Reply by xitters** [reply]
> Try adding the exact same tracks to the queue directly (I.e. not via the party interface) and see what happens then.



The same thing happens, it doesn't always happen, now I tried skipping ahead almost to the end of a song and adding 2 more quickly afterward and it skipped them

**Reply by OzGav** [reply]
Then this has nothing to do with Party. Firstly, make sure you are on the latest version of MA and if the problem persists ask your own question in the Q&A

**Reply by xitters** [reply]
> Then this has nothing to do with Party. Firstly, make sure you are on the latest version of MA and if the problem persists ask your own question in the Q&A

I think disable the option allow croaafades between tracks of different samples rates on sonos resolves the skips issues. But I've noticed that if a song is added by party mode plugging right after the previous one ends, it can get skipped. Is this possible?

**Reply by MarvinSchenkel** [reply]
> > Then this has nothing to do with Party. Firstly, make sure you are on the latest version of MA and if the problem persists ask your own question in the Q&A
> 
> 
> 
> I think disable the option allow croaafades between tracks of different samples rates on sonos resolves the skips issues. But I've noticed that if a song is added by party mode plugging right after the previous one ends, it can get skipped. Is this possible?

Yes that's possible. Sonos buffers songs very aggressively so whenever you have a queue of track A and B and you add C as "next Track" when A is done playing, track B has already been buffered by Sonos. 

We should add them after B instead of skipping them though.

**Reply by apophisnow** [reply]
I was pretty sure I fixed this, so it would add the request as next +1 position if a track was buffered already. Maybe we had a regression, or a sonos specific thing? If we get additional reports I can certainly look into it again.

**Reply by MarvinSchenkel** [reply]
Also saw that your fix is only included in 2.8.2, so might be fixed already

**Comment by UKenGB** [comment]
How is party mode added to the LH menu?

It was there after I updated to 2.8.2, but having rearranged that menu, it's now gone and not listed as an option to add.

Not needed right now, but I'd like to know how to show and hide it as necessary.

**Reply by OzGav** [reply]
Do that via MA SETTINGS >> USER INTERFACE

**Reply by UKenGB** [reply]
Ah, but as I said, it is NOT there in the pop-up menu as an option to add.

I wanted to re-arrange the items on the left and the only way to do that is to add them from that menu in the order you want (except 'Search' which is forced to the top) and since 'Party' was not there to remove individually, I hit the x to remove them all and added them back in the order I wanted.

Later I realised that if I did want to add 'Party' back, I could not as it is not there as an option in the menu. Hence my question.

**Reply by OzGav** [reply]
This is fixed in the nightly. New beta comes out toorrow.

**Reply by UKenGB** [reply]
👍

**Comment by teek541** [comment]


**Reply by OzGav** [reply]
Feature requests need to be raised separately.

**Reply by teek541** [reply]
Corrected

**Reply by OzGav** [reply]
No sorry. You raise them as separate discussions entirely with the feature request type and we then see how many votes they get and implement accordingly. So I will delete the three posts below.

edit: if they are closely related you can combine some feature requests together.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
